src/TurbulenceClosures/turbulence_closure_implementations/ri_based_vertical_diffusivity.jl

```diff
@@ -189,21 +189,30 @@ end
     entraining = (!convecting) & (N²_above < 0)
 
     # Convective adjustment diffusivity
-    κᶜ = ifelse(convecting, κᶜ, zero(grid))
+    κᶜᵃ = ifelse(convecting, κᶜᵃ, zero(grid))
 
     # Entrainment diffusivity
-    κᵉ = ifelse(Qᵇ > 0, Cᵉ * Qᵇ / N², zero(grid))
-    κᵉ = ifelse(entraining, Cᵉ, zero(grid))
+    κᵉⁿ = ifelse(Qᵇ > 0, Cᵉⁿ * Qᵇ / N², zero(grid))
+    κᵉⁿ = ifelse(entraining, Cᵉⁿ, zero(grid))
 
     # Shear mixing diffusivity and viscosity
     Ri = Riᶜᶜᶠ(i, j, k, grid, velocities, tracers, buoyancy)
 
     τ = taper(tapering, Ri, Ri₀, Riᵟ)
-    κ★ = κ₀ * τ
-    ν★ = ν₀ * τ
+    κᶜ★ = κ₀ * τ
+    κᵘ★ = ν₀ * τ
 
-    @inbounds diffusivities.κ[i, j, k] = κᶜ + κᵉ + κ★
-    @inbounds diffusivities.ν[i, j, k] = ν★
+    # Previous diffusivities
+    κᶜ = diffusivities.κᶜ
+    κᵘ = diffusivities.κᵘ
+
+    # New diffusivities
+    κᶜ⁺ = κᶜᵃ + κᵉⁿ + κᶜ★
+    κᵘ⁺ = κᵘ★
+
+    # Update by averaging in time
+    @inbounds κᶜ[i, j, k] = 1/2 * (κᶜ[i, j, k] + κᶜ⁺)
+    @inbounds κᵘ[i, j, k] = 1/2 * (κᵘ[i, j, k] + κᵘ⁺)
```

Comment:
this is the mean with the previous value... so it's not a "time average", right? there is not time-scale involved here... or to phrase it better it's a time average over a timescale that is proportional with the tilmestep. The results will become time-step dependent, right?

Reply:
Yes, it's a time-average proportional to the time step. It's like forward-Euler-stepping this equation

```julia
dt κᶜ = (κᶜ⁺ - κᶜ) / 2Δt
```

you might be right that results are sensitive to time-step... i'll test that

Reply:
A counter-argument to my argument is that not doing anything at all still produces things that depend on the time-step.

Reply:
So we could fix the time-scale by writing

```julia
d/dt κᶜ = (κᶜ⁺ - κᶜ) / τ
```

Then we update with

```julia
κᶜ = (1 - Δt/τ) κᶜ + Δt / τ * κᶜ⁺
```

Reply:
But actually backward Euler is better:

```julia
κᶜ = (ϵ * κᶜ⁺ + κᶜ) / (1 + ϵ)
```

Reply:
Great!!